lib/matplotlib/collections.py

```diff
@@ -1128,7 +1128,8 @@ def legend_elements(self, prop="colors", num="auto",
             raise ValueError("Valid values for `prop` are 'colors' or "
                              f"'sizes'. You supplied '{prop}' instead.")
 
-        fmt.set_bounds(func(u).min(), func(u).max())
+        fmt.axis.set_view_interval(func(u).min(), func(u).max())
+        fmt.axis.set_data_interval(func(u).min(), func(u).max())
```

Comment:
Do we really want to call `func` four times over?

Reply:
indeed, fixed